xapian-applications/omega/omindex.cc

```diff
@@ -193,7 +193,7 @@ index_directory(const string &path, const string &url_, size_t depth_limit,
 	    try {
 		switch (d.get_type()) {
 		    case DirectoryIterator::DIRECTORY: {
-			char * dot_ptr = strrchr(d.leafname, '.');	
+			char * dot_ptr = strrchr(d.leafname(), '.');	
```

Comment:
You need to check your code compiles locally at the very, very minimum before pushing changes (if calling `strrchr()` on a function pointer compiled locally, there's something very wrong with your compiler).  You should really be checking that the appropriate test suite passes too, since a patch which causes tests to fail can't be merged.

There's little point us reviewing code when it won't even compile - you're wasting our limited patch reviewing time.  Also every change you push to a PR against Xapian currently triggers 12 different builds, which are all on donated VMs - by triggering pointless builds like these you delay other useful builds.

Reply:
Oh, I really apologize. I didn't released it. Since my my laptop isn't repaired yet, I don't have a dedicated system to work on. I thought to save the work on repo itself each time. I'll use some other way now on.

Reply:
Turns out I can deploy Travis CI and AppVeyor in my repo with few clicks :)

Reply:
That avoids us getting an email about code that isn't useful to review, but it still means running 12 builds on the same donated infrastructure, and that means less build capacity is available to others using travis and appveyor.

It's really not sensible to use CI as a replacement for having your own a machine to build on.